pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -542,10 +542,14 @@ def _binary_op_method_timedeltalike(op, name):
 
         elif hasattr(other, 'dtype'):
             # nd-array like
-            if other.dtype.kind not in ['m', 'M']:
+            if other.dtype.kind not in ['m', 'M', 'O']:
                 # raise rathering than letting numpy return wrong answer
                 return NotImplemented
-            return op(self.to_timedelta64(), other)
+            try:
+                converted_other = other.astype('datetime64[ns]')
```

Comment:
Converting is wrong here, since it could be mixed, eg [Timestamp, Timedelta].

Reply:
The mixed test I added passes but I'm guessing this case is why I'm getting problems in other tests? Is there another approach I should take?

Reply:
Ah there was a bug in my test. Still looking around but definitely curious on another approach. I was thinking I could iterate over the entire array and do piecemeal conversions but that also seems wrong.